data/src/main/java/com/m3u/data/service/internal/PlayerManagerImpl.kt

```diff
@@ -169,14 +169,16 @@ class PlayerManagerImpl @Inject constructor(
             val licenseKey = channel.licenseKey.orEmpty()
             channelRepository.reportPlayed(channel.id)
             val playlist = playlistRepository.get(channel.playlistUrl)
+            val kodiUrlOptions = channelUrl.readKodiUrlOptions()
+            val userAgent = kodiUrlOptions[KodiAdaptions.HTTP_OPTION_UA] ?: playlist?.userAgent
```

Comment:
Place user-agent code to a function, because the tryPlay function as below will use wrong user-agent.

Reply:
done